modular/gater/metadata_handler.go

```diff
@@ -277,13 +275,19 @@ func (g *GateModular) listObjectsByBucketNameHandler(w http.ResponseWriter, r *h
 		ContinuationToken:     continuationToken,
 	}
 
-	respBytes, err = xml.Marshal(grpcResponse)
-	if err != nil {
-		log.CtxErrorw(reqCtx.Context(), "failed to get user buckets", "error", err)
+	if format == "json" {
+		respBytes, err = json.Marshal(grpcResponse)
+		w.Header().Set(ContentTypeHeader, ContentTypeJSONHeaderValue)
+		w.Write(respBytes)
 		return
 	}
 
-	respBytes = processObjectsXmlResponse(respBytes, grpcResponse.Objects)
+	respBytes, err = xml.Marshal((*GfSpListObjectsByBucketNameResponse)(grpcResponse))
```

Comment:
Does the `Forced Typecasting` necessary here?

Reply:
fixed